components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/GatewayArtifactsMgtDAO.java

```diff
@@ -49,9 +49,10 @@ public static GatewayArtifactsMgtDAO getInstance() {
      * @param tenantDomain - Tenant domain of the API
      * @throws APIManagementException if an error occurs
      */
-    public void addGatewayPublishedAPIDetails(String APIId, String APIName, String version, String tenantDomain)
+    public boolean addGatewayPublishedAPIDetails(String APIId, String APIName, String version, String tenantDomain)
```

Comment:
Even though the function signature has been changed to return a boolean, no action is being done by checking the return type at call time.

Reply:
We are checking the return flag in the test method https://github.com/wso2/carbon-apimgt/pull/9002/files#diff-5d5216ef9cc19eb82608ade2690638cdR106